app/service/agent_svc.py

```diff
@@ -72,7 +72,14 @@ def __init__(self, untrusted_timer):
             update_data = dict(last_seen=now, pid=pid, ppid=ppid)
             if agent[0]['trusted']:
                 update_data['last_trusted_seen'] = now
-            await self.data_svc.update('core_agent', 'paw', paw, data=update_data)
+            await self.get_service('data_svc').update('core_agent', 'paw', paw, data=update_data)
+            incoming_executors =[]
+            for i,e in enumerate(executors):
+                incoming_executors.append(dict(executor=e, preferred=1 if i == 0 else 0))
```

Comment:
I'd like to see this logic grouped together like the existing create_agent logic so it offers a single responsibility look & feel. how to do this? I would create an update_agent() function here on the data_svc and see if you can fill it in.

good work on the delete portion-- that is easy to overlook but you caught it.

Reply:
I can do that. I was trying to stay away from making modifications in data_svc. I will rework this!